bulk_sync_into_benchling/bulk_sync_into_benchling.py

```diff
@@ -0,0 +1,205 @@
+import json
+
+import click
+import requests
+import time
+
+
+class BadRequestException(Exception):
+    def __init__(self, message, rv):
+        super(BadRequestException, self).__init__(message)
+        self.rv = rv
+
+
+def api_get(domain, api_key, path):
+    url = "https://{}/api/v2/{}".format(domain, path)
+    rv = requests.get(url, auth=(api_key, ""))
+    if rv.status_code >= 400:
+        raise BadRequestException(
+            "Server returned status {}. Response:\n{}".format(
+                rv.status_code, json.dumps(rv.json())
+            ),
+            rv,
+        )
+    return rv.json()
+
+
+def api_post(domain, api_key, path, body):
+    url = "https://{}/api/v2/{}".format(domain, path)
+    rv = requests.post(url, json=body, auth=(api_key, ""))
+    if rv.status_code >= 400:
+        raise BadRequestException(
+            "Server returned status {}. Response:\n{}".format(
+                rv.status_code, json.dumps(rv.json())
+            ),
+            rv,
+        )
+    return rv.json()
+
+
+def wait_on_task_response(domain, api_key, task_resource):
+    task_id = task_resource["taskId"]
+    while True:
+        task_response = api_get(domain, api_key, "tasks/{}".format(task_id))
+        status = task_response["status"]
+        if status == "RUNNING":
+            time.sleep(10)
+        else:
+            if status == "FAILED":
+                return "FAILED", task_response["message"], task_response["errors"]
+            elif status == "SUCCEEDED":
+                return "SUCCEEDED", task_response["response"]
+
+
+@click.command()
+@click.option(
+    "--domain",
+    help="Domain name of your Benchling instance, e.g. example.benchling.com",
+    required=True,
+)
+@click.option("--api-key", help="Your API key", required=True)
+@click.option(
+    "--folder-id", help="ID of a folder to create the antibody in", required=True
+)
+@click.option("--registry-id", help="ID of the Benchling Registry", required=True)
+@click.option("--antibody-schema-id", help="ID of the Antibody schema", required=True)
+@click.option(
+    "--chain-schema-id",
+    help="ID of the Chain schema (Must be an AA-Sequence)",
+    required=True,
+)
+@click.argument("json_file_to_import", type=click.File("r"))
+def main(
+    domain,
+    api_key,
+    antibody_schema_id,
+    registry_id,
+    chain_schema_id,
+    folder_id,
+    json_file_to_import,
+):
+    antibodies_json = json.loads(json_file_to_import.read())
```

Comment:
This is a common benchling naming quirk, but IMO it's pretty confusing. These objects are now loaded and are no longer json.

Reply:
Fixed.